parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestInputOutputFormatWithBloomFilter.java

```diff
@@ -0,0 +1,249 @@
+
+    Configuration conf = new Configuration();
+
+    // Don't use a cached FS with a different block size
+    conf.set("fs.file.impl.disable.cache", "true");
+
+    // Disable summary metadata.
+    conf.set("parquet.summary.metadata.level", "none");
+    conf.set("parquet.example.schema", PARQUET_TYPE.toString());
+
+    {
+      conf.set("parquet.bloom.filter.column.names", "uuid,value");
+      conf.set("parquet.bloom.filter.expected.ndv", String.join(",",EXPECTED_DISTINCT_NUMBERS));
+
+      Job writeJob = new Job(conf, "writeWithBloomFilter");
+      writeJob.setInputFormatClass(TextInputFormat.class);
+      TextInputFormat.addInputPath(writeJob, new Path(inputFile.toString()));
+      writeJob.setOutputFormatClass(ParquetOutputFormat.class);
+      writeJob.setMapperClass(TestInputOutputFormatWithBloomFilter.Writer.class);
+      writeJob.setNumReduceTasks(0); // write directly to Parquet without reduce
+      ParquetOutputFormat.setWriteSupportClass(writeJob, GroupWriteSupport.class);
+      ParquetOutputFormat.setBlockSize(writeJob, 2048);
+      ParquetOutputFormat.setPageSize(writeJob, 512);
+      ParquetOutputFormat.setEnableDictionary(writeJob, false);
+      ParquetOutputFormat.setMaxPaddingSize(writeJob, 1023); // always pad
+      ParquetOutputFormat.setOutputPath(writeJob, tempPath);
+
+      // Check Bloom filter parameters
+      HashMap expectedParameter = new HashMap<String,Long>(){{
+        put("uuid",200L);
+        put("value",50L);
+      }};
+      Assert.assertTrue("Bloom filter parameter should be correct inside Hadoop conf.", expectedParameter.equals(ParquetOutputFormat.getBloomFilterColumnExpectedNDVs(conf)));
+
+      waitForJob(writeJob);
+
+    }
+
+    // Check bloom filter for each block and column
+    Path path = getDataPath(tempFolder);
+
+    ParquetMetadata readFooter = ParquetFileReader.readFooter(conf, path);
+    ParquetFileReader r = new ParquetFileReader(conf,path,readFooter);
+
+    int startBlock = 0;
+    for(BlockMetaData block : readFooter.getBlocks()){
+      int rowCount = (int) block.getRowCount();
+      int endBlock = rowCount + startBlock;
+      int columnIndex = 0;
+      for(ColumnChunkMetaData columnMetaData : block.getColumns()) {
+
+        Long expectedDistinctNumbers = Long.parseLong(EXPECTED_DISTINCT_NUMBERS[columnIndex]);
+
+        // Build bloom filter from the generated data
+        BloomFilter expectedBloomFilter = buildBloomFilter(columnMetaData, startBlock, endBlock,columnIndex, expectedDistinctNumbers);
+
+        // Read bloom filter from the parquet file
+        BloomFilterReader bloomFilterReader = r.getBloomFilterDataReader(block);
+        BloomFilter bloomFilter = bloomFilterReader.readBloomFilter(columnMetaData);
+
+        Assert.assertTrue("BloomFilter bitset should be the same", bloomFilter.equals(expectedBloomFilter));
+
+        columnIndex++;
+      }
+      startBlock = endBlock;
+    }
+
+  }
+
+  private String generateData(){
```

Comment:
I think we can reuse this logic: https://github.com/apache/parquet-mr/pull/725/files#diff-18d0dc81eac31c2e5454354594ce7058R35-R42

Reply:
I agree!